### PR TITLE
#167276162 Create inputForm Field molecule

### PR DESCRIPTION
**Pull request by akinyeleolat**
#### What does this PR do?

- This PR sets up an Input Form field molecule with styleguide

#### Description of task to be completed?

- setup input form field molecule
- setup styleguidist for the input form field molecule
- write tests for the input form field molecule
- refactor divider to display inline
- add web pack molecule module resolver
- add jest config molecule module resolver
- add molecule sections to style guide configuration

#### How should this be manually tested?

  - $ git clone https://github.com/andela/king-kong-ah-frontend.git
  - $ npm install
  - $ npm test

#### Any background context you want to provide?

- You would need npm to run this application

#### What are the relevant pivotal tracker stories?

- PT Story link - [#167276162](https://www.pivotaltracker.com/story/show/167276162)

#### Screenshots (if appropriate)
Screenshot 2019-07-13 at 7 25 18 PM

### CODE REVIEW

**Review by frankly034 — APPROVED** [review]

**Review by Sojisoyoye — APPROVED** [review]

File: src/components/UI/molecules/InputFormField/InputFormField.jsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  onChange,
+  errorMessage,
+  inputWidth,
+  borderRadius,
+  type,
+  name,
+  label,
+  display,
+}) => {
+  return (
```
Comment (benfluleck) [inline src/components/UI/molecules/InputFormField/InputFormField.jsx]: There should be a space here above the `return`

File: src/components/UI/molecules/InputFormField/InputFormField.jsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        name={name}
+        type={type}
+        placeholder={placeholder}
+        value={value}
+        inputWidth={inputWidth}
+        borderRadius={borderRadius}
+        error={errorMessage}
+        onChange={onChange} />
+      {errorMessage
+        && <span><Text color='error' fontSize='small' display='block'>
```
Comment (benfluleck) [inline src/components/UI/molecules/InputFormField/InputFormField.jsx]: I expect the color to be red not error

File: src/components/UI/molecules/InputFormField/InputFormField.jsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  borderRadius,
+  type,
+  name,
+  label,
+  display,
+}) => {
+  return (
+
+    <FlexContainer display={display}>
+      <label htmlFor={name}>
```
Comment (benfluleck) [inline src/components/UI/molecules/InputFormField/InputFormField.jsx]: Do a Label molecule. Doing this here means any time the label is edited we need to edit here.

File: src/components/UI/molecules/InputFormField/InputFormField.spec.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+describe('Input Form Fields', () => {
+  it('should render a input field component', () => {
+    const { container } = setUp();
+    expect(container).toBeTruthy();
+  });
+
+  it('should render a input password field component', () => {
+    const { container } = setUp({ type: 'password' });
+    expect(container).toBeTruthy();
```
Comment (benfluleck) [inline src/components/UI/molecules/InputFormField/InputFormField.spec.js]: This is not the correct test I expect a test for one of the props

File: src/components/UI/molecules/InputFormField/InputFormField.spec.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  });
+
+
+  it('should render a input field component with error message', () => {
+    const { container } = setUp({ errorMessage: 'first name is required' });
```
Comment (benfluleck) [inline src/components/UI/molecules/InputFormField/InputFormField.spec.js]: This is not the correct test I expect a test for one of the props

**Review by benfluleck — CHANGES_REQUESTED** [review]

**Review by benfluleck — APPROVED** [review]

**Pull request merged**